### PR TITLE
Adding code-server and LE certs to ansible-tower-advanced config

### PR DESCRIPTION
**Pull request by goetzrieger**
#### SUMMARY

Some changes to the _ansible-tower-advanced_ config we are preparing to use for the Summit 2021 Advanced Tower lab.

#### CHANGES

- Changes to _vscode-server_ role to make it work with RHEL 8 (repo/packages)
- New role _automation_controller_issue certs_ to configure the Tower nodes with LE certs. Basically transplanted from the _issue_certs_ role from the Ansible Workshops provisioner (Sean C)
- Changes to _ansible-tower-advanced_ config to use both roles

@IPvSean @tonykay @rhjcd @ericzolf

### CODE REVIEW

File: ansible/configs/ansible-tower-advanced/files/ansible-tower-advanced_sample_vars.yml [hunk 1]
```diff
@@ -0,0 +1,168 @@
+# TIP: CHANGEME needs to be adapted once for each developer, CHANGEME-ec2 needs to be adapted for each AWS sandbox
```
Comment (ericzolf) [inline ansible/configs/ansible-tower-advanced/files/ansible-tower-advanced_sample_vars.yml:1]: The whole file seems a duplicate of the sample_vars.yml file.

File: ansible/configs/ansible-tower-advanced/software.yml [hunk 2]
```diff
@@ -7,17 +7,43 @@
...
+- hosts: all
+  gather_facts: yes 
+  tasks: 
+    - name: Dump all variables
+      local_action: 
+        module: template
+        src: dumpall.j2 
+        dest: /tmp/ansible_dump.all
+      run_once: true
+      when: dump_vars|default(false)|bool == true
```
Comment (ericzolf) [inline ansible/configs/ansible-tower-advanced/software.yml:19]: you don't need/want the == true

File: ansible/roles/automation_controller_issue_cert/files/lets-encrypt-x3-cross-signed.key [hunk 3]
```diff
@@ -0,0 +1,27 @@
+-----BEGIN CERTIFICATE-----
```
Comment (ericzolf) [inline ansible/roles/automation_controller_issue_cert/files/lets-encrypt-x3-cross-signed.key:1]: Urgh, I don't think there should be a key in the repository, it should actually be ignored in .gitignore.

Comment (ericzolf) [reply ansible/roles/automation_controller_issue_cert/files/lets-encrypt-x3-cross-signed.key:1]: Or is it the public key of Let's Encrypt?

**Review by ericzolf — COMMENTED** [review]
Few review comments, I can't merge anyway :smirk:

File: ansible/configs/ansible-tower-advanced/files/ansible-tower-advanced_sample_vars.yml (hunk 1 above)
Comment (goetzrieger) [reply ansible/configs/ansible-tower-advanced/files/ansible-tower-advanced_sample_vars.yml:1]: Never mind, just tried a cleaner vars sample file. Feel free to replace. ;)

File: ansible/roles/automation_controller_issue_cert/files/lets-encrypt-x3-cross-signed.key (hunk 3 above)
Comment (goetzrieger) [reply ansible/roles/automation_controller_issue_cert/files/lets-encrypt-x3-cross-signed.key:1]: It's the cross signed intermediate, this is from the Ansible WS provisioner role. Just noticed on https://letsencrypt.org/certificates/ it's actually retired, sigh. Still works, though.

**Review by rhjcd — APPROVED** [review]

**Pull request merged**